projects/plugins/jetpack/3rd-party/class-salesforce-lead-form.php

```diff
@@ -0,0 +1,142 @@
+<?php
+/**
+ * Salesforce Lead Form using Jetpack Contact Forms.
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack;
+
+/**
+ * Class Salesforce_Lead_Form
+ *
+ * Hooks on Jetpack's Contact form to send form data to Salesforce.
+ */
+class Salesforce_Lead_Form {
+	/**
+	 * Salesforce_Contact_Form constructor.
+	 * Hooks on `grunion_after_feedback_post_inserted` action to send form data to Salesforce.
+	 */
+	public static function initialize() {
+		add_action( 'grunion_after_feedback_post_inserted', array( __CLASS__, 'process_salesforce_form' ), 10, 4 );
+	}
+
+	/**
+	 * Process Salesforce Lead forms
+	 *
+	 * @param int   $post_id - the post_id for the CPT that is created.
+	 * @param array $fields - Grunion_Contact_Form_Field array.
+	 * @param bool  $is_spam - marked as spam by Akismet(?).
+	 * @param array $entry_values - extra fields added to from the contact form.
+	 *
+	 * @return null|void
+	 */
+	public static function process_salesforce_form( $post_id, $fields, $is_spam, $entry_values ) {
+		// if spam (hinted by akismet?), don't process
+		if ( $is_spam ) {
+			return;
+		}
+
+		$blocks = parse_blocks( get_the_content() );
+
+		$filtered_blocks = self::get_salesforce_contact_form_blocks( $blocks );
+
+		// no contact-form blocks with salesforceData and organizationId, move on
+		if ( empty( $filtered_blocks ) ) {
+			return;
+		}
+
+		// more than one form on post, skipping process
+		if ( count( $filtered_blocks ) > 1 ) {
+			return;
+		}
+
+		$attrs           = $filtered_blocks[0]['attrs']['salesforceData'];
+		$organization_id = $attrs['organizationId'];
+		// Double sanity check: no organization ID? Abort.
+		if ( empty( $organization_id ) ) {
+			return;
+		}
+
+		$keyed_fields = array_map(
+			function ( $field ) {
+				return $field->value;
+			},
+			$fields
+		);
+
+		// this is yet TBD, campaign IDs are hard to get from SF app/UI, but if
+		// the user filled it, then send as API field Campaign_ID
+		if ( ! empty( $attrs['campaignId'] ) ) {
+			$keyed_fields['Campaign_ID'] = $attrs['campaignId'];
+		}
+
+		// add post/page URL as lead_source
+		$keyed_fields['lead_source'] = $entry_values['entry_permalink'];
+		$keyed_fields['oid']         = $organization_id;
+
+		// we got this far, try and send it. Need to check for errors on submit
+		try {
+			self::send_to_salesforce( $keyed_fields );
+		} catch ( \Exception $e ) {
+			// phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_trigger_error
+			trigger_error( sprintf( 'Jetpack Form: Sending lead to Salesforce failed: %s', esc_html( $e->getMessage() ) ) );
```

Comment:
Wouldn't we also get this message as a fatal error if we just don't catch the thrown exception? Were you planning to add a handler of some sort here later?

Reply:
Yes, but I think I can tackle this as we continue to work on customizations (next iteration expected as soon as this is merged). @zinigor do you have any tips regarding error logging (kibana or otherwise)?